projects/sandbox/mdc/mdc/main.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import time
+from pathlib import Path
+from typing import List, Optional
+
+import h5py
+from mdc.callback import Callback
+from mdc.data import ChunkedSegmentLoader, batch_chunks
+from typeo import scriptify
+
+from aframe.logging import configure_logging
+from hermes.aeriel.client import InferenceClient
+
+
+@scriptify
+def main(
+    ip: str,
+    model_name: str,
+    data_dir: Path,
+    start: int,
+    output_dir: Path,
+    sample_rate: float,
+    inference_sampling_rate: float,
+    ifos: List[str],
+    batch_size: int,
+    integration_window_length: float,
+    cluster_window_length: float,
+    fduration: float,
+    throughput: float,
+    chunk_size: float,
+    sequence_id: int,
+    model_version: int = -1,
+    log_file: Optional[Path] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Perform inference using Triton on a directory
+    of timeseries files, using a particular set of
+    interferometer time shifts. Network outputs will
+    be saved both as-is and using local integration.
+
+    Args:
+        ip:
+            The IP address at which a Triton server
+            hosting the indicated model is running
+        model_name:
+            The name of the model to which to make
+            inference requests
+        data_dir:
+            Directory containing input files representing
+            timeseries on which to perform inference.
+            Each HDF5 file in this directory will be used
+            for inference.
+        write_dir:
+            Directory to which to save raw and locally
+            integrated network outputs.
+        sample_rate:
+            Rate at which input timeseries data has
+            been sampled.
+        inference_sampling_rate:
+            The rate at which to sample windows for
+            inference from the input timeseries.
+            Corresponds to the sample rate of the
+            output timeseries.
+        batch_size:
+            The number of subsequent windows to
+            include in a single batch of inference.
+        window_length:
+            Length of the window over which network
+            outputs should be locally integrated,
+            specified in seconds.
+        max_shift:
```

Comment:
Doesn't matter, but this makes me realize we need to change this in the `infer` project

Reply:
Ah yeah good catch, we'll need to do a thorough doc review before publication where we weed out stuff like this